src/plugin-utils.ts

```diff
@@ -0,0 +1,129 @@
+import path from 'path';
+import { Required } from 'utility-types';
+import { Store, Reporter } from 'gatsby';
+import checkHasYarn from 'has-yarn';
+
+import { PluginOptions, DeprecatedPluginOptions } from './types';
+import { formatLanguage } from './common';
+
+export type RequiredPluginOptions = Required<Omit<PluginOptions, keyof DeprecatedPluginOptions>>;
+
+interface RequirePluginOptionsFn {
+  (
+    options: unknown,
+    props: {
+      reporter: Reporter,
+      store: Store,
+    },
+  ): RequiredPluginOptions;
+}
+export const requirePluginOptions: RequirePluginOptionsFn = (
+  options,
+  {
+    store,
+    reporter,
+  },
+) => {
+  const hasYarn = checkHasYarn();
+
+  const { program } = store.getState();
+  const basePath = program.directory as string;
+
+  // There are no required properties (yet), so must be compatible.
+  const pluginOptions = options as PluginOptions;
+
+  const {
+    language = 'typescript',
+    includeResolvers = false,
```

Comment:
Would it be better to let user specify additional @graphql-codegen plugin themselves, instead of doing a toggle here? i.e If someone needs the resolver plugin, they could require it in gatsby-config then add it to a 'codegenPlugins' array

Reply:
I thought about project goals. It's not about writing a codegen configuration preset, but about integrating development with types to be more natural in Gatsby.

I think hiding codegen interfaces from the user would be better.

So If the user needs to configure codegen, for example for Apollo client, the user can still use codegen CLI, but without worrying about Gatsby at all.